tensordict/dynamic_shaped_tensor.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+import torch
+
+from tensordict import MemmapTensor
+
+
+class DynamicShapeTensor:
```

Comment:
This is slightly at odds with the name of the submodule `dynamic_shaped_tensor` vs. `DynamicShapeTensor`.

Reply:
ah right! let's go for shaped no?
Any other idea for the name by the way?

Reply:
I'm happy with "shaped".

Only other thing that sprung to mind was something like "jagged" or "ragged", but I think you just run the risk of confusion with `KeyedJaggedTensor` etc.?